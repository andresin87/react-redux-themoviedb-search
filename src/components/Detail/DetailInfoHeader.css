.DetailInfoHeaderTagline {
  margin-left: 10px;
  font-style: 'oblique';
}

/**
 * Resposiveness for a long name movie.
 * For instance, The Assassination of Jesse James by the Coward Robert Ford
 */
@media (max-width: 1500px) {
  .DetailInfoHeader {
    max-width: 500px;
  }
}

@media (max-width: 1250px) {
  .DetailInfoHeader {
    max-width: 450px;
  }
}

@media (max-width: 1100px) {
  .DetailInfoHeader {
    max-width: 400px;
  }
}

@media (max-width: 1050px) {
  .DetailInfoHeader {
    max-width: 100%;
  }
}

@media (max-width: 775px) {
  .DetailInfoHeader {
    max-width: 500px;
  }
}

@media (max-width: 650px) {
  .DetailInfoHeader {
    max-width: 400px;
  }
}

@media (max-width: 520px) {
  .DetailInfoHeader {
    max-width: 100%;
  }
}
