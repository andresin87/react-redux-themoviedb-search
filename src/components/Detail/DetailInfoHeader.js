import React, { PureComponent, Fragment } from 'react';

class DetailInfoHeader extends PureComponent {
  getTagline = tagline => {
    if (!tagline || tagline === '') return null;
    else {
      return (
        <h4 style={{ marginLeft: '10px', fontStyle: 'oblique' }}>
          {' '}
          â€” {tagline}
        </h4>
      );
    }
  };

  render() {
    const { title, tagline } = this.props;
    return (
      <Fragment>
        <h2>
          <strong>{title}</strong>
        </h2>
        {this.getTagline(tagline)}
      </Fragment>
    );
  }
}

export default DetailInfoHeader;
