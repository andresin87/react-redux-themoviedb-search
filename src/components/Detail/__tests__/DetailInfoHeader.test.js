import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import DetailInfoHeader from '../DetailInfoHeader';

describe('DetailInfoHeader component', () => {
  const title = 'Fight Club';
  const tagline = 'Mischief. Mayhem. Soap.';
  const component = <DetailInfoHeader title={title} tagline={tagline} />;

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Snapshot tagline', () => {
    it('No tagline', () => {
      const component = <DetailInfoHeader title={title} />;
      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Tagline is null', () => {
      const component = <DetailInfoHeader title={title} tagline={null} />;
      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Tagline is empty string', () => {
      const component = <DetailInfoHeader title={title} tagline={''} />;
      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
