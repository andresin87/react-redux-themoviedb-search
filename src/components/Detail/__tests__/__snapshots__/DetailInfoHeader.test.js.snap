// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailInfoHeader component Snapshot matchs 1`] = `
Array [
  <h2>
    <strong>
        Fight Club
    </strong>
</h2>,
  <h4
    style={
        Object {
            "fontStyle": "oblique",
            "marginLeft": "10px",
          }
    }
>
     
    â€” 
    Mischief. Mayhem. Soap.
</h4>,
]
`;

exports[`DetailInfoHeader component Snapshot tagline No tagline 1`] = `
<h2>
  <strong>
    Fight Club
  </strong>
</h2>
`;

exports[`DetailInfoHeader component Snapshot tagline Tagline is empty string 1`] = `
<h2>
  <strong>
    Fight Club
  </strong>
</h2>
`;

exports[`DetailInfoHeader component Snapshot tagline Tagline is null 1`] = `
<h2>
  <strong>
    Fight Club
  </strong>
</h2>
`;
